sync/rpc/server.go

```diff
@@ -1232,10 +1263,33 @@ func (s *SyncAgentServer) syncFilesFromHealthyReplica(fromServer string, disks [
 		if err := s.syncFile(fromServer, disk+".meta"); err != nil {
 			return err
 		}
+
+		s.RebuildInfo.Lock()
+		s.RebuildInfo.processed++
+		s.RebuildInfo.Unlock()
 	}
+
+	s.finishRebuild()
+
 	return nil
 }
 
+func (s *SyncAgentServer) prepareRebuild() {
+	s.RebuildInfo.Lock()
+	defer s.RebuildInfo.Unlock()
+
+	s.RebuildInfo.IsRebuilding = true
```

Comment:
Do we need to save this flag in struct `SyncAgentServer`? Just like `PrepareRestore()`, `PreparePurge()`